### ISSUE TITLE
Checkout Error when Stripe Payment Method Incorrect

### ISSUE THREAD

**Issue by lin-d-hop** [openfoodfoundation/openfoodnetwork#4713]
## Description

When creating Payment Methods it is easy to create a Stripe payment method with no credentials. If you accidentally select the Stripe payment method type instead of ... say ... Cash/Cheque ... then you will be unlikely to notice that you haven't created it correctly.

Following this if you finish setting up your shop you'll get a 500 error at checkout and probably (if you are me) have absolutely no idea why the checkout is failing and panic and create S1 bugs and declare an emergency and storm around Slack being angry and .... you get the idea.

It would be nice if the user was informed that they had created a broken payment method earlier. It would also be nice if said broken payment method failed gracefully instead of nasty slug attacks just when you are checking out your lettuce.

## Expected Behavior

If I select Stripe as the Payment Method Provider then click Create I want to see an error and have this payment method creation fail.

## Actual Behaviour

I select Stripe as the Payment Method Provider then click Create and the payment method successfully creates.... then fails at checkout

## Steps to Reproduce

1. Create a new Payment method with Provider Stripe. Do not connect a Stripe account
2. Create the Order Cycle as usual
3. Create basket and go to Checkout
4. DEATH BY SLUG

## Workaround

Look in the logs. See the cause. Fix it.
If a user does this accidentally we'd be on S1 alert until the cause was found.

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: 2.7
* Browser name and version: Chrome
* Operating System and version (desktop or mobile): Ubuntu

## Possible Fix

Papercut candidate

**Comment by luisramos0** [comment]
the root cause of the error 500 is https://github.com/openfoodfoundation/openfoodnetwork/issues/4044

**Comment by luisramos0** [comment]
this needs re-testing, reevaluation after #4758 at least it will be lower priority I believe.

**Comment by luisramos0** [comment]
I did the "re evaluation" myself now. I updated the title of the issue. It's now an S4 :+1:

**Comment by Erioldoesdesign** [comment]
Is there a part of this papercut (that might be outside papercut territory) that is about UI error validation?

**Comment by luisramos0** [comment]
Hi @Erioldoesdesign I am not sure I understand your question. What do you mean by "UI error validation"? This issue will show an error to the user in one new case (stripe credentials not defined). There is a  standard way to show form errors in the backoffice right now: it's that red box with a list of issues.

**Comment by Erioldoesdesign** [comment]
Sorry, When I said 'UI error validation' I should have said 'inline UI error validation' which you can see an example of [here](https://baymard.com/blog/inline-form-validation) 
It's also not helpful to this issue now that I have looked at the payment methods settings deeper. I'm still getting used to where to raise ideas related to issues that aren't always immediately 'part of' the issue.

**Comment by luisramos0** [comment]
ah, ok, I get it.
I dont think that's a extra to an issue. I think that specific one should be raised as a separate issue by itself. Currently I think we have very few inline form validations currently.
